src/lib/layout/aem-model-provider/aem-model-provider.component.ts

```diff
@@ -10,13 +10,15 @@
  * governing permissions and limitations under the License.
  */
 
-import { Component, Input, ViewChild } from '@angular/core';
-import { ModelManager } from '@adobe/aem-spa-page-model-manager';
+import { Component, Input, Output, ViewChild, EventEmitter } from '@angular/core';
+import { ModelManager, PathUtils } from '@adobe/aem-spa-page-model-manager';
 import { AEMComponentDirective } from '../aem-component.directive';
+import { Constants } from '../constants';
+import { Utils } from '../utils';
 
 @Component({
   selector: 'aem-model-provider,[aemModelProvider]',
-  template: `<ng-container aemComponent [cqItem]='cqItem' [cqPath]='cqPath' [itemName]='itemName'></ng-container>`
+  template: `<ng-container *ngIf="cqItemLoaded | async" aemComponent [cqItem]='cqItem' [cqPath]='cqPath' [itemName]='itemName'></ng-container>`
```

Comment:
What happens if we don't add this if here? I am thinking, will we break someones code if we are not rendering this container (as in he had a fallback, ghost loading or so, and now it's not going to be rendered)?

Reply:
For basic SPA, the condition is resolved by default so it shouldn't change the existing behavior in any way. For remote SPA, the container should only be loaded after the model has been fetched. But you raise an interesting point - I'll look into error handling as well when for when the model fetch fails.

Reply:
If the container is loaded without the model being fetched - it would still result in the component not being rendered as it is dependent on the data. It wouldn't break the page - that particular component simply wouldn't be rendered. There is already a listener for cqPath so whenever the path is updated, model fetch is done again and it would render as intended. The if condition here is simply to prevent premature attempts to render the component until the data is available.